packages/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -230,7 +231,7 @@ export const getCellGroupingCriteria = ({
   if (groupingRule.groupingValueGetter) {
     key = groupingRule.groupingValueGetter(row[groupingRule.field] as never, row, colDef, apiRef);
   } else {
-    key = row[groupingRule.field] as GridKeyValue | null | undefined;
+    key = getRowValue(row, colDef, apiRef) as GridKeyValue | null | undefined;
```

Comment:
When the row grouping is initially enabled, the public methods aren't yet assigned to `apiRef`.
I extracted the `getRowValue` method to a function so I can still call it here.
This doesn't change the fact that the `apiRef` passed to `valueGetter` won't have public methods assigned yet.

This makes me think that we should assign the methods (at least some of them) *before* initialization 🤔
https://github.com/mui/mui-x/blob/27997634d9d919e7e17bdaff24453d486304ef46/packages/x-data-grid-premium/src/DataGridPremium/useDataGridPremiumComponent.tsx#L165
WDYT?